### ISSUE TITLE
[BUG] Error with duplicates

### ISSUE THREAD

**Issue by DarKDinDoN** [trioderegion/warpgate#8]
Hello,

I have an error with duplicates.

When I run this:
```javascript
await warpgate.spawn(
        'Dancing Lights',
        {token: {name: `${item.data.name} (${actor.data.name})`}},
        {},
        {duplicates: 4}
    )
```

I get this error:
![image](https://user-images.githubusercontent.com/1687854/129488732-65baddb2-1f40-4824-9f65-13538d7fd3d2.png)
I'm using a lot of module wrapping methods with LibWrapper.

Ran the same script without all the modules and without LibWrapper, got this error:
![image](https://user-images.githubusercontent.com/1687854/129488777-c849c072-e7a1-44dc-aeb1-891cbd4370eb.png)

**Comment by trioderegion** [comment]
odd. i thought i fixed this prior to release. I'll look into it

**Comment by trioderegion** [comment]
Hmm. I can't seem to reproduce. Could you export your Dancing Lights actor for me to look at?

**Comment by DarKDinDoN** [comment]
@trioderegion 
here is the Dancing Lights item: https://we.tl/t-gI6l4SEtAh
Yes, i'm using Item Macro.

**Comment by trioderegion** [comment]
Thanks, it looks like its somehow related to the new dnd5e version 🤔 Weird as I don't deal with system specific operations.

**Comment by trioderegion** [comment]
Could you test the exact same script, but using this actor (named 'wildcard')? It is from dnd5e 1.4.1 and might have different data.

I was unable to reproduce the error with this actor, but could on a new one in 1.4.2.
[fvtt-Actor-dancing-lights.zip](https://github.com/trioderegion/warpgate/files/6989412/fvtt-Actor-dancing-lights.zip)

**Comment by trioderegion** [comment]
Update: 

It appears to be due to a known issue with semaphore handling on database updates, which can cause race conditions with back to back token creation -> token actor update calls.

I have improved the code in general while hunting this bug, but the end result (for now) is to slap a small wait between my token creation and actor update. I will make this wait an adjustable value to help accommodate higher ping times.

Its ugly...but I think its the best I have right now.  Look for a 1.3.1 soon.

**Comment by DarKDinDoN** [comment]
@trioderegion Thank you for the research!
I'll test on 1.3.1 and let you know if it's working correctly.

**Comment by trioderegion** [comment]
Closed in 1.4.0
